packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -59,5 +78,25 @@ export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
         throwNotFoundError: true,
       });
     },
+    createPurpose: async (
+      purposeSeed: m2mGatewayApi.PurposeSeed,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Purpose> => {
```

Comment:
For consistency with other things already merged, let's not use arrow function props here, but proper functions 
(same for other methods in this file, missed it in previous reviews)

```suggestion
    async createPurpose(
      purposeSeed: m2mGatewayApi.PurposeSeed,
      { logger, headers }: WithLogger<M2MGatewayAppContext>
    ): Promise<m2mGatewayApi.Purpose> {
```